lib/kernel/src/socket.erl

```diff
@@ -3958,11 +3983,11 @@ With arguments `Msg` and `Flags`; equivalent to
 With arguments `Data` and `Cont`; equivalent to
 [`sendmsg(Socket, Data, Cont, infinity)`](`sendmsg/4`) *since OTP 24.0*.
 """.
--spec sendmsg(Socket :: term(), Msg :: term(), Flags :: list())
+-spec sendmsg(Socket :: socket(), Msg :: msg_send(), Flags :: list())
              -> _;
-             (Socket :: term(), Data :: term(), Cont :: select_info())
+             (Socket :: socket(), Data :: msg_send() | erlang:iovec(), Cont :: select_info())
              -> _;
-             (Socket :: term(), Msg :: term(), Timeout :: term())
+             (Socket :: socket(), Msg :: msg_send(), Timeout :: infinity)
```

Comment:
`_` = term() is too wide return type. Can we get the old types back?

(Or use `dynamic()` to not break type-checking)

Reply:
For now, maybe `dynamic()` is better than `term()` and for most purposes we really mean `dynamic()` and not the top type. So I think that's ok